Source/UnitEx.cpp

```diff
@@ -176,7 +176,7 @@ struct OP2Unit
 	unsigned int flags;
 	char bayWeaponCargo[6];
 	char unknown6;
-	char unknownCargo;
+	char conVecCargoTurret; // Turret of a guard post stored in a conVec

```

Comment:
Do we want to switch all the char values to std::uint8_t or std::int8_t unless they are specifically meant to be a char? 

If this name looks okay, it needs to be pushed into the SVN repo ollydbg section. I determined the functionality by comparing a convec with a guard post loaded and right after it was dumped. This value is the map_id of the turret type. The value stays set after dumping, but the cargo of the convec is no longer a guard post.

Reply:
Using `std::uint8_t` sounds like a sensible idea. I would support doing that. As this affects many fields, perhaps it should get its own issue.

----

Thank you for figuring out the field to look for. To confuse matters, there is already a field `short weaponCargo`, though I believe that weapon value is only for military vehicles and guard posts. I had initially assumed that field would be relevant here, though clearly it's not, and using that field would be akin to putting the weapon on the Convec itself.

Just a reminder the meaning of fields is dependent on the unit type. The OllyDbg comments text files lump a lot of unit data together, though there is some partial splitting it out into unit subclasses. It was a work in progress that was never finished. I don't believe the Convec class is well mapped yet.

If you look at the `OP2Unit` struct above, you can see that it clearly mixes fields from very different unit types. That leads to a somewhat non-sense mismash of fields. For accuracy, they really need to be split out into a `union` of `struct`/`class`. I recognize that would be a separate issue.

Reply:
I opened a char to uint8_t issue. 

I understand what you are saying about OP2Unit. Although I think it is beyond my current skill to untangle even if I wanted to put the time into it. I wouldn't mind putting some time into learning about unions in the near future.

Reply:
Yeah, it's going to be a mess to untangle and document the unit hierarchy and all the associated fields. What I'm suggesting is necessarily going to be a long term thing. It's quite likely going to entail and number of branches over a long period of time, rather than a single PR to get everything into the right place.

I guess for now, the text file documentation can have new unit types added. New fields we figure out that are specific to a unit can then be documented with the unit type.

We should maybe find a way to indicate when only partial field information is known. In particular, knowing the parent class and size of the parent class is very useful information. That gives some indication of class layout for where unit specific fields of the derived class start. It's also nice to know at a glance when there are gaps between known fields, sized of unknown fields, or having a bound on the size of a class within the hierarchy. Though considering all unit records take 120 bytes, it's hard to known a closer max size for a class, unless you know of a more derived class that has member fields that wouldn't exist in the parent class.

I'm getting off topic.

Also, this might related more strongly to the OP2Internal project. There is quite a bit of overlap in what these two projects cover. Actually some of the text file documentation was removed after the information was encoded into OP2Internal in code form. Maybe someday the OP2Internal and HFL projects can be merged.